### PR TITLE
Add open_file, works on macOS, safer, Add buttons to open settings file and config directory

### PR DESCRIPTION
**Pull request by def-**
Screenshot 2021-12-19 at 01 23 12
## Checklist

- [x] Tested the change ingame
- [x] Provided screenshots if it is a visual change
- [x] Tested in combination with possibly related configuration options
- [ ] Written a unit test if it works standalone, system.c especially
- [x] Considered possible null pointers and out of bounds array indexing
- [x] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/base/system.h [hunk 1]
```diff
@@ -1816,18 +1816,27 @@ int net_socket_read_wait(NETSOCKET sock, int time);
...
+
 	Parameters:
 		link - The link to open in a browser.
-	
+
 	Returns:
 		Returns 1 on success, 0 on failure.
-	
-	Remarks:
-		This may not be called with untrusted input or it'll result in arbitrary code execution.
```
Comment (heinrich5991) [inline src/base/system.h:1827]: I don't think the Windows version is safe yet. It only has a single "command line string" that gets passed, the receiving program will use libc's command line parsing to get the arguments. By passing a filename with spaces, you should be able to escape and pass parameters to `open`.

File: src/base/system.cpp [hunk 2]
```diff
@@ -3666,13 +3666,26 @@ int open_link(const char *link)
 	MultiByteToWideChar(CP_UTF8, 0, link, -1, wBuffer, sizeof(wBuffer) / sizeof(WCHAR));
 	return (uintptr_t)ShellExecuteW(NULL, L"open", wBuffer, NULL, NULL, SW_SHOWDEFAULT) > 32;
 #elif defined(CONF_PLATFORM_LINUX)
-	char aBuf[512];
-	str_format(aBuf, sizeof(aBuf), "xdg-open %s >/dev/null 2>&1 &", link);
-	return system(aBuf) == 0;
+	const int pid = fork();
+	if(pid == 0)
+		execl("/usr/bin/env", "env", "xdg-open", link, nullptr);
```
Comment (heinrich5991) [inline src/base/system.cpp:3671]:
```suggestion
		execl("xdg-open", link, nullptr);
```
Double `env`. None required, I think.

**Review by heinrich5991 — CHANGES_REQUESTED** [review]
Looks like a fine change.

Need clarification for the removed security notice before merging.

File: src/base/system.h (hunk 1 above)
Comment (def-) [reply src/base/system.h:1827]: Ah, too bad.

File: src/base/system.cpp (hunk 2 above)
Comment (def-) [reply src/base/system.cpp]: Done without env. The double open is required though since the first argument has to be the name of the program.

**Review by heinrich5991 — APPROVED** [review]

**Comment by heinrich5991** [comment]
bors r+

Thanks.

**Pull request merged**